src/VisualStudio/InteractiveServices/Interactive/VsInteractiveWindowPackage.cs

```diff
@@ -30,22 +28,24 @@ internal abstract partial class VsInteractiveWindowPackage<TVsInteractiveWindowP
         private IComponentModel _componentModel;
         private TVsInteractiveWindowProvider _interactiveWindowProvider;
 
-        protected override void Initialize()
+        protected override async Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
         {
-            base.Initialize();
+            await base.InitializeAsync(cancellationToken, progress).ConfigureAwait(true);
+
+            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
             // Load the Roslyn package so that its FatalError handlers are hooked up.
-            var shell = (IVsShell)this.GetService(typeof(SVsShell));
+            var shell = (IVsShell)await GetServiceAsync(typeof(SVsShell)).ConfigureAwait(true);
```

Comment:
so, there was this long discussion on whether GetServiceAsync can return null or not. and how to handle it. and looks like we don't check that anywhere we changed to GetServiceAsync.

unlike before where GetService can only return null when service is not available, now it can return null on cancellation, so unless we are okay to crash on shutdown, we should now guard from it?

Reply:
➡️ Resolved in cf727ada0b58dc689af808263d95ab1cd3d529fb